server.js

```diff
@@ -15,6 +16,82 @@ app.get("/", function (request, response) {
 
 // TODO add your routes and helper functions here
 
-const listener = app.listen(process.env.PORT, function () {
+//Route to Post new Booking
+
+app.post("/booking", (req, res) => {
+  const newBooking = {
+    id: req.body.id,
+    title: req.body.title,
+    firstName: req.body.firstName,
+    surname: req.body.surname,
+    email: req.body.email,
+    roomId: req.body.roomId,
+    checkInDate: req.body.checkInDate,
+    checkOutDate: req.body.checkOutDate,
+  };
+
+  // usually we do the validation in both the front end and in the backend , I have this validation
+  //for the sake of simplicity
+
+  if (
+    !newBooking.id ||
+    !newBooking.title ||
+    !newBooking.firstName ||
+    !newBooking.surname ||
+    !newBooking.email ||
+    !newBooking.roomId ||
+    !newBooking.checkInDate ||
+    !newBooking.checkOutDate
+  )
+    return res.status(400).send("Please Enter All data");
+
+  bookings.push(newBooking);
+  res.json(bookings);
+});
+
+// Route to return all bookings
+app.get("/bookings", function (req, res) {
+  //Route to retuen bookings in a specific date
+  const searchDate = req.query.date;
+  console.log(searchDate);
+  if (searchDate) {
+    // res.json(searchDate);
+    res.json(
+      bookings.filter(
+        (booking) =>
+          booking.checkInDate === searchDate ||
+          booking.checkOutDate === searchDate
+      )
+    );
+  } else {
+    res.json(bookings);
+  }
+});
+
+// Route to read booking of a specified id arguments
+app.get("/booking/:id", (req, res) => {
+  const queriedId = parseInt(req.params.id);
+  const index = bookings.map((booking) => booking.id).indexOf(queriedId);
```

Comment:
this is good :) There are additional array methods available in Node.js (and most browsers) that make this easier - see `find` and `findIndex`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex